atomicapp/nulecule/lib.py

```diff
@@ -94,7 +101,11 @@ def get_provider(self, provider_key=None, dry=False):
         # If provider_key isn't provided via CLI, let's grab it the configuration
         if provider_key is None:
             provider_key = self.config.get(GLOBAL_CONF)[PROVIDER_KEY]
-        provider_class = self.plugin.getProvider(provider_key)
+        try:
+            provider_class = self.plugin.getProvider(provider_key)
+        except NuleculeException as e:
+            logger.error(e)
+            sys.exit(1)
```

Comment:
@dustymabe You're right! We do most exception handling in ``atomicapp/cli/main.py``. We don't need to put a ``try-except`` block around ``provider_class = self.plugin.getProvider(provider_key)`` because the ``NuleculeException`` raised by ``getProvider`` will be automatically handled in ``cli/main.py``.

@surajssd can you remove the ``try...except`` block here?